packages/host/app/components/operator-mode/stack.gts

```diff
@@ -22,48 +29,52 @@ interface Signature {
       stackItem: StackItem,
     ) => void;
     setupStackItem: (
-      stackItem: StackItem,
-      clearSelections: () => void,
-      doWithStableScroll: (changeSizeCallback: () => Promise<void>) => void,
-      doScrollIntoView: (selector: string) => void,
+      model: StackItem,
+      componentAPI: StackItemComponentAPI,
     ) => void;
   };
   Blocks: {};
 }
 
 export default class OperatorModeStack extends Component<Signature> {
-  private closeAnimation = new WeakMap<StackItem, () => void>();
+  private stackItemComponentAPI = new WeakMap<
+    StackItem,
+    StackItemComponentAPI
+  >();
 
   dismissStackedCardsAbove = task(async (itemIndex: number) => {
     let itemsToDismiss: StackItem[] = [];
     for (let i = this.args.stackItems.length - 1; i > itemIndex; i--) {
       itemsToDismiss.push(this.args.stackItems[i]);
     }
 
-    // do closing animation on last item
-    const lastItem = this.args.stackItems[this.args.stackItems.length - 1];
-    const closeAnimation = this.closeAnimation.get(lastItem);
-    if (closeAnimation) {
-      await closeAnimation();
-    }
+    // do closing animation on each closed item
+    await Promise.all(
+      itemsToDismiss.map((i) => {
+        const componentAPI = this.stackItemComponentAPI.get(i);
+        if (!componentAPI) {
+          return Promise.resolve();
+        }
+        return componentAPI.startAnimation('closing');
```

Comment:
the type for componentAPI.startAnimation says that it returns void. But this code is acting like it returns a promise. Figure out which is right and bring them into alignment.

Reply:
Thanks for the feedback, I have adjusted accordingly